### ISSUE TITLE
[FR]: Apostrophe Scanning

### ISSUE THREAD

**Issue by bensnhunt** [yomidevs/yomitan#1630]
Some words and phrases fail to be scanned, even where they have entries in the dictionary, due to an apostrophe being in the middle of them. 

Even after disabling the recommended settings where `j'` `l'` and `d'` are replaced with empty strings, and scanning per character, we are still unable to select certain strings.

e.g.
- `d'après` according to (in the opinion of)
- `tout d'un coup` all of a sudden
- `n'importe quoi` nonesense
- `d'accord` in agreement
etc. etc.

Edit: fix typo `apres` to `après`

**Comment by StefanVukovic99** [comment]
Have you also tried tweaking the `Advanced > Text Scanning > Sentence termination characters` setting to not terminate on apostrophes?

**Comment by bensnhunt** [comment]
Yes, setting `Text Parsing > Sentence termination characters` to `None` still fails to scan past the apostrophe.

**Comment by StefanVukovic99** [comment]
Is this using `kty-fr-en` or another dictionary?

**Comment by StefanVukovic99** [comment]
On French apostrophe usage:

- what kinds of apostrophe are used and how commonly? I gather `’` is most correct, `'` is common in informal contexts; are there others?
- have I correctly understood that there should be no space after the apostrophe but it's common enough that we should support scanning with the space there?

**Comment by StefanVukovic99** [comment]
kty-fr-en currently has some issues with apostrophes, should be fixed in the next version. With that and #1635, the examples look like this:
[Screencast from 12-01-2024 05:30:29 PM.webm](https://github.com/user-attachments/assets/d3f3f575-7962-4981-b4e9-22b7fdd2d7d4)

**Comment by bensnhunt** [comment]
> Is this using `kty-fr-en` or another dictionary?

Using both `kty-fr-en` and `kty-fr-fr`; both contain entries with apostrophes.

> what kinds of apostrophe are used and how commonly? I gather ’ is most correct, ' is common in informal contexts; are there others?

From experience it seems like `'` i.e. `U+0027` is used most commonly, although with most things on the internet people will use different characters like you mention: `’` i.e. `U+ 2019`. But there's no difference between them in terms of what they represent in French, just different substitutes for the same character.

 > have I correctly understood that there should be no space after the apostrophe but it's common enough that we should support scanning with the space there?

In practically all cases, there is no space following the apostrophe, in human or automatically generated text. If there is a space following, then it's maybe best to treat it as the end of a quotation or something similar.


In summary, apostrophe gets used to show the loss of a vowel in French e.g. `de > d'`. But because `de` is very commonly used in derivation and thus into dictionary words e.g. `pomme de terre` (which yomitan is able to capture), it often shows up as `d'` e.g. `prise d'acte` (which yomitan is not able to capture).

This will of course cause issues if we are using word-level scanning, since `j'` commonly attaches to the verb e.g. `j'attends` which means we can't scan the verb itself `attends`. For character level, however, it's fine.

**Comment by bensnhunt** [comment]
> kty-fr-en currently has some issues with apostrophes, should be fixed in the next version. With that and #1635, the examples look like this: [Screencast from 12-01-2024 05:30:29 PM.webm](https://github.com/user-attachments/assets/d3f3f575-7962-4981-b4e9-22b7fdd2d7d4)

This fix looks great, functions as expected :)
